src/test/java/com/onfido/integration/ReportSchemasTest.java

```diff
@@ -52,17 +52,18 @@ public void schemaOfFacialSimilarityPhotoReportIsValid() throws Exception {
                 .reportNames(Arrays.asList(ReportName.FACIAL_SIMILARITY_PHOTO_FULLY_AUTO)));
 
     FacialSimilarityPhotoFullyAutoReport facialSimilarityPhotoFullyAutoReport =
-        onfido.findReport(check.getReportIds().get(0)).getFacialSimilarityPhotoFullyAutoReport();
+        (FacialSimilarityPhotoFullyAutoReport)
+            repeatRequestUntilStatusChanges(
+                "findReport", new Object[] {check.getReportIds().get(0)}, COMPLETE, 10, 1000);
 
-    int iteration = 0;
-    while (!facialSimilarityPhotoFullyAutoReport.getStatus().equals(COMPLETE)) {
-      if (iteration > 10) {
-        Assertions.fail("Report did not complete in time");
-      }
-      iteration += 1;
-      Thread.sleep(1000);
-      facialSimilarityPhotoFullyAutoReport =
-          onfido.findReport(check.getReportIds().get(0)).getFacialSimilarityPhotoFullyAutoReport();
-    }
+    Assertions.assertEquals(COMPLETE, facialSimilarityPhotoFullyAutoReport.getStatus());
+    Assertions.assertEquals(
+        ReportName.FACIAL_SIMILARITY_PHOTO_FULLY_AUTO,
+        facialSimilarityPhotoFullyAutoReport.getName());
+    Assertions.assertEquals(check.getId(), facialSimilarityPhotoFullyAutoReport.getCheckId());
+    Assertions.assertNotNull(facialSimilarityPhotoFullyAutoReport.getBreakdown());
+    Assertions.assertNotNull(facialSimilarityPhotoFullyAutoReport.getResult());
+    Assertions.assertNotNull(
+        facialSimilarityPhotoFullyAutoReport.getAdditionalProperties().get("properties"));
```

Comment:
I don't see too much the interest of this test. If properties field is set for given report type, it should be accessible by calling `.getProperties()` method... 🤔